src/api.cc

```diff
@@ -419,10 +418,11 @@ CheckInResult AkliteClient::CheckInLocal(const LocalUpdateSource* local_update_s
 
   LOG_INFO << "The local TUF repo has been successfully updated";
   LOG_INFO << "Searching for matching TUF Targets...";
-  auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, client_->tags, secondary_hwids_);
+  auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, {}, secondary_hwids_);
```

Comment:
>Having a tag configured on the device tells TUF to filter for specific targets.

It is not quite correct statement. The device tag tells to filter for specific set of TUF metadata - <tag> -> [timestamp, snapshot, targets].

And, the latter commit does verify if the bundle's tag matches the device's one. So, this check is kind of redundant.

Reply:
I've noticed that second check.
Given that, a check in this function makes no harm... it is in fact a kind of a sanity check, which is rather logical.

I usually prefer to have too many checks than too little checks, simply because it is more robust.

Reply:
I am actually kind on the fence about this check, that's why I implemented in the way allowing its easy removal. I think we need to discuss it offline, including customer support team (this repo is public).